pkgs/applications/science/biology/tandem-aligner/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, zlib
+}:
+
+stdenv.mkDerivation {
+  pname = "TandemAligner";
+  version = "unstable-2022-09-17";
+
+  src = fetchFromGitHub {
+    owner = "seryrzu";
+    repo = "tandem_aligner";
+    rev = "ac6004f108ad20477045f4d0b037d96051a9df70";
+    hash = "sha256-iMDj1HZ8LzmZckuAM3lbG3eSJSd/5JGVA6SBs7+AgX8=";
+  };
+
+  sourceRoot = "source/tandem_aligner";
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ zlib ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp src/projects/tandem_aligner/tandem_aligner $out/bin
+    runHook postInstall
+  '';
+
+  doCheck = true;
+
+  # adapted from target test_launch in Makefile
+  checkPhase = ''
+    runHook preCheck
+    mkdir -p $TMPDIR/test_launch
+    src/projects/tandem_aligner/tandem_aligner \
+      --first $src/tandem_aligner/test_dataset/first.fasta \
+      --second $src/tandem_aligner/test_dataset/second.fasta \
+      -o $TMPDIR/test_launch \
+      --debug
+    grep -q "Thank you for using TandemAligner!" $TMPDIR/test_launch/tandem_aligner.log
+    diff $TMPDIR/test_launch/cigar.txt $src/tandem_aligner/test_dataset/true_cigar.txt
```

Comment:
This wouldn't be run if grep failed

Reply:
Yes, these are [upstream sanity checks](https://github.com/seryrzu/tandem_aligner/blob/ac6004f108ad20477045f4d0b037d96051a9df70/tandem_aligner/Makefile#L22-L23) that the program behaves as expected. If any of the two would fail, something is wrong. But maybe you are hinting at something different?

Reply:
I assume the diff should help debug a failure? This won't be the case here since the build will stop before that. And the diff command would only fail if it can't generate the diff, not if there is a diff.

Reply:
No, the diff is just there to ensure that the test run did produce exactly what is expected, namely the contents of `true_cigar.txt`.